samples/nrf9160/lwm2m_client/fota.rst

```diff
@@ -0,0 +1,301 @@
+If the client is not able to satisfy such a request, the update process will fail with the Update Result resource set to ``13``.
+
+Library support
+***************
+
+Advanced Firmware update is part of the :ref:`lib_lwm2m_client_utils` library and is designed to support updating of modem and application binary updates.
+
+Configuration option
+====================
+
+Enable the following Kconfig options to use advanced FOTA:
+
+.. code-block:: console
+
+   CONFIG_LWM2M_CLIENT_UTILS=y
+   CONFIG_LWM2M_CLIENT_UTILS_ADV_FIRMWARE_UPDATE_OBJ_SUPPORT=y
+   CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_UPDATE_OBJ_SUPPORT=y
+   CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT=n
+   CONFIG_FOTA_CLIENT_AUTOSCHEDULE_UPDATE=n
+
+The :file:`overlay-adv-firmware.conf` configuration file includes these options.
+
+Initializing
+============
+
+The Advanced Firmware Update object shares the same API with the standard LwM2M Firmware Update object as implemented in :ref:`lib_lwm2m_client_utils`.
+The Advanced Firmware Update functionality is implemented in :file:`lwm2m_firmware.c` and :file:`lwm2m_adv_firmware.c`.
+
+The Advanced Firmware Update object is initialized by calling the :c:func:`lwm2m_init_firmware` function and by confirming the currently booted application image as valid by calling the :c:func:`lwm2m_init_image` function.
+The result of validating the image marks the FOTA process as completed and is reported to the server.
+The modem FOTA process is automatically validated.
+
+LwM2M Client utilities library setup
+====================================
+
+By default, the Advanced Firmware Update object supports two instances that are configured in the following way:
+
++-------------+-------------+---------------------------------------------------------------------+
+| Instance ID | Owner       | DFU types                                                           |
++=============+=============+=====================================================================+
+| 0           | Application | DFU_TARGET_IMAGE_TYPE_MCUBOOT                                       |
++-------------+-------------+---------------------------------------------------------------------+
+| 1           | Modem       | DFU_TARGET_IMAGE_TYPE_MODEM_DELTA, DFU_TARGET_IMAGE_TYPE_FULL_MODEM |
++-------------+-------------+---------------------------------------------------------------------+
+
+Advanced Firmware Update helper script
+--------------------------------------
+
+For working with the `AVSystem Coiote server <Coiote Device Management_>`_, a helper script has been provided.
+To automate the firmware update, you can use the script :file:`fota.py` that is available in the :file:`samples/nrf9160/lwm2m_client/scripts/` folder.
+For firmware updates, the :file:`/fota.py` file supports ``update`` and ``upload`` commands.
+
++-----------------+-------------------------------+-------------+-------------------------------------------------+
```

Comment:
@umapraseeda and @divipillai 
Shuold we also change this table to command or keep it?

Reply:
Good point. Yes, we can. Added a suggestion.